addons/auth_signup/controllers/main.py

```diff
@@ -86,6 +93,11 @@ def web_auth_reset_password(self, *args, **kw):
             except Exception as e:
                 qcontext['error'] = str(e)
 
+        elif 'signup_email' in qcontext:
+            user_ids = request.env['res.users'].sudo().search([('partner_id.email', '=', qcontext.get('signup_email'))])
+            if user_ids and all(u.state != 'new' for u in user_ids):
+                return request.redirect('/web/login?%s' % url_encode({'login': qcontext.get('signup_email'), 'redirect': '/web'}))
```

Comment:
careful that you can have a login different than the email of the user (e.g. faba instead of faba@odoo.com on our database)

shouldn't you use the login here, and not the email address ?